Lack of column names quoting in Phoenix connector's underlying queries

```
CREATE TABLE t("column-with-hyphen" varchar);

SELECT "column-with-hyphen" FROM t

2020-04-17T11:12:45.7820921Z Caused by: org.apache.phoenix.exception.PhoenixParserException: ERROR 601 (42P00): Syntax error. Encountered "-" at line 1, column 110.
2020-04-17T11:12:45.7824069Z 	at org.apache.phoenix.exception.PhoenixParserException.newException(PhoenixParserException.java:33)
2020-04-17T11:12:45.7826666Z 	at org.apache.phoenix.parse.SQLParser.parseStatement(SQLParser.java:111)
2020-04-17T11:12:45.7829968Z 	at org.apache.phoenix.jdbc.PhoenixStatement$PhoenixStatementParser.parseStatement(PhoenixStatement.java:1644)
2020-04-17T11:12:45.7832746Z 	at org.apache.phoenix.jdbc.PhoenixStatement.parseStatement(PhoenixStatement.java:1727)
2020-04-17T11:12:45.7835428Z 	at org.apache.phoenix.jdbc.PhoenixStatement.execute(PhoenixStatement.java:1819)
2020-04-17T11:12:45.7837960Z 	at io.prestosql.plugin.jdbc.BaseJdbcClient.execute(BaseJdbcClient.java:824)
2020-04-17T11:12:45.7839196Z 	... 16 more
2020-04-17T11:12:45.7841684Z Caused by: org.apache.phoenix.shaded.org.antlr.runtime.NoViableAltException
2020-04-17T11:12:45.7844612Z 	at org.apache.phoenix.parse.PhoenixSQLParser.column_name(PhoenixSQLParser.java:2966)
2020-04-17T11:12:45.7847892Z 	at org.apache.phoenix.parse.PhoenixSQLParser.column_def(PhoenixSQLParser.java:4458)
2020-04-17T11:12:45.7850839Z 	at org.apache.phoenix.parse.PhoenixSQLParser.column_defs(PhoenixSQLParser.java:4404)
2020-04-17T11:12:45.7884303Z 	at org.apache.phoenix.parse.PhoenixSQLParser.create_table_node(PhoenixSQLParser.java:1236)
2020-04-17T11:12:45.7884460Z 	at org.apache.phoenix.parse.PhoenixSQLParser.oneStatement(PhoenixSQLParser.java:868)
2020-04-17T11:12:45.7884736Z 	at org.apache.phoenix.parse.PhoenixSQLParser.statement(PhoenixSQLParser.java:524)
2020-04-17T11:12:45.7884883Z 	at org.apache.phoenix.parse.SQLParser.parseStatement(SQLParser.java:108)
2020-04-17T11:12:45.7884995Z 	... 20 more
```

--- a/presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java
+++ b/presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java
@@ -143,6 +143,7 @@ import static org.apache.hadoop.hbase.HConstants.FOREVER;
 import static org.apache.phoenix.coprocessor.BaseScannerRegionObserver.SKIP_REGION_BOUNDARY_CHECK;
 import static org.apache.phoenix.util.PhoenixRuntime.getTable;
 import static org.apache.phoenix.util.SchemaUtil.ESCAPE_CHARACTER;
+import static org.apache.phoenix.util.SchemaUtil.getEscapedArgument;
 
 public class PhoenixClient
         extends BaseJdbcClient
@@ -389,7 +390,7 @@ public class PhoenixClient
                     typeStatement += " not null";
                     pkNames.add(columnName);
                 }
-                columnList.add(format("%s %s", columnName, typeStatement));
+                columnList.add(format("%s %s", getEscapedArgument(columnName), typeStatement));
             }
 
             ImmutableList.Builder<String> tableOptions = ImmutableList.builder();

--- a/presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixDistributedQueries.java
+++ b/presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixDistributedQueries.java
@@ -140,10 +140,10 @@ public class TestPhoenixDistributedQueries
     }
 
     @Override
-    public void testColumnName(String columnName)
+    protected boolean isColumnNameRejected(Exception exception, String columnName, boolean delimited)
     {
-        // TODO (https://github.com/prestosql/presto/issues/3466) Phoenix generally lacks quoting in underlying queries
-        throw new SkipException("TODO");
+        // TODO This should produce a reasonable exception message like "Invalid column name". Then, we should verify the actual exception message
+        return columnName.equals("a\"quote");
     }
 
     @Override

